### PR TITLE
fix(api): allow labeling and versioning for module labware

### LINKED ISSUE #4595: ModuleContext.load_labware() is missing useful parameters from ProtocolContext().load_labware
## overview

The definition of ``ProtocolContext.load_labware()`` is thus:

```
    @requires_version(2, 0)
    def load_labware(
            self,
            load_name: str,
            location: types.DeckLocation,
            label: str = None,
            namespace: str = None,
            version: int = None,
    ) -> Labware:
       etc
```

However, the definition of ``ModuleContext.load_labware()`` is thus:

``` @requires_version(2, 0)
    def load_labware(self, name: str) -> Labware:
        """ Specify the presence of a piece of labware on the module.

        :param name: The name of the labware object.
        :returns: The initialized and loaded labware object.
        """
        lw = load(
            name, self._geometry.location,
            bundled_defs=self._ctx._bundled_labware)
        return self.load_labware_object(lw)
```

It makes sense that in the latter there is no ``location`` parameter. However, the omission of the ``label``, ``namespace``, and ``version`` parameters appears to be an oversight.

## current behavior

See above

## steps to reproduce

See above 

## expected behavior

The signature of ``ModuleContext.load_labware()`` should differ from ``ProtocolContext.load_labware()`` only where semantically required (i.e.: the omission of ``location``).

### LINKED ISSUE #4612: _extra_labware missing from ModuleContext.load_labware
## overview

``ModuleContext.load_labware`` is currently thus:

```
    @requires_version(2, 0)
    def load_labware(self, name: str) -> Labware:
        """ Specify the presence of a piece of labware on the module.

        :param name: The name of the labware object.
        :returns: The initialized and loaded labware object.
        """
        lw = load(
            name, self._geometry.location,
            bundled_defs=self._ctx._bundled_labware)
        return self.load_labware_object(lw)
```

Notice that ``_bundled_labware`` from the ``ProtocolContext`` is passed to the ``load``, but the ``_extra_labware`` is not.

In contrast, in ``ProtocolContext.load_labware``, _both_ the ``_bundled_labware`` and the ``_extra_labware`` are passed:

```
        labware_def = get_labware_definition(
            load_name, namespace, version,
            bundled_defs=self._bundled_labware,
            extra_defs=self._extra_labware)
        return self.load_labware_from_definition(
            labware_def, location, label)
```

Is this an oversight? Or is it intentional behavior?

### PR DESCRIPTION
**Pull request by ahiuchingau**
## overview
Currently, `ModuleContext.load_labware` is missing a few important parameters, such as `label`, `namespace`, and `version`. Also, `_extra_labware` should also be passed into the `load` function in `ModuleContext.load_labware`, as pointed out in #4612.

This PR adds:
- `label`, `namespace`, and `version` to `ModuleContext.load_labware`
- `_extra_labware`to be passed to the load function in `ModuleContext.load_labware`
- `label` to `ModuleContext.load_labware_from_definition`
- bump the max supported API version!
- docstring fixup for `ProtocolContext.load_labware_from_definition`
- more tests!

closes #4595, closes #4612

### CODE REVIEW

File: api/src/opentrons/protocol_api/contexts.py [hunk 1]
```diff
@@ -310,7 +314,7 @@ def load_labware(
             location: types.DeckLocation,
             label: str = None,
             namespace: str = None,
-            version: int = None,
+            version: int = 1,
```
Comment (sfoster1) [inline api/src/opentrons/protocol_api/contexts.py]: We should keep this at `None` and continue establishing in code that it defaults to 1. This is actually set in the `labware._get_standard_labware_definition`.

File: api/src/opentrons/protocol_api/contexts.py [hunk 2]
```diff
@@ -1957,29 +1961,49 @@ def load_labware_object(self, labware: Labware) -> Labware:
         return mod_labware
 
     @requires_version(2, 0)
-    def load_labware(self, name: str) -> Labware:
+    def load_labware(
+            self,
+            name: str,
+            label: str = None,
+            namespace: str = None,
+            version: int = 1,
```
Comment (sfoster1) [inline api/src/opentrons/protocol_api/contexts.py]: Same as `ProtocolContext.get_labware` - the version should default to `None`, and that it means 1 called out in the docs.

File: api/src/opentrons/protocol_api/contexts.py [hunk 3]
```diff
@@ -1957,29 +1961,49 @@ def load_labware_object(self, labware: Labware) -> Labware:
         return mod_labware
 
     @requires_version(2, 0)
-    def load_labware(self, name: str) -> Labware:
+    def load_labware(
```
Comment (sfoster1) [inline api/src/opentrons/protocol_api/contexts.py]: This is our first change that requires a protocol api minor version bump! It adds new behavior, and should be version 2.1. Requiring users to set `"apiLevel": "2.1"` means that they get saner errors if they run their protocols on older robot software versions. We should call out the new parameters as only being available in version 2.1 in the docstring.

There are kind of two ways we could handle this:
- In the body of the function, check if the version is less than 2.1 and raise an error if any of the parameters are specified. The error should be the same one raised by the `requires_version` decorator.
- Add capability to the `requires_version` decorator to specify some parameters as only existing at a different api level from the rest of the function. Then we can check the parameters in the call.

This runs the risk of making that decorator quite unwieldy, but we could keep the version checking logic out of the body of the function.

Either way, _something_ needs to complain when you try to use the new parameters at 2.0.

File: api/src/opentrons/protocol_api/contexts.py [hunk 4]
```diff
@@ -1957,29 +1961,49 @@ def load_labware_object(self, labware: Labware) -> Labware:
...
+            name, self._geometry.location, label, namespace, version,
             bundled_defs=self._ctx._bundled_labware)
         return self.load_labware_object(lw)
 
     @requires_version(2, 0)
     def load_labware_from_definition(
-            self, definition: Dict[str, Any]) -> Labware:
+            self,
+            definition: Dict[str, Any],
+            label: str = None) -> Labware:
```
Comment (sfoster1) [inline api/src/opentrons/protocol_api/contexts.py]: the `label` parameter here should be gated behind api level 2.1.

File: api/src/opentrons/protocol_api/contexts.py (hunk 3 above)
Comment (rgatkinson) [reply api/src/opentrons/protocol_api/contexts.py]: Comments from the peanut gallery: I'm really not sure that you need to _fail_ if the new params are used at 2.0: there's no semantic conflict in question; there's no ambiguity as to intent. 

What bumping the minor version _does_ give is a way for the programmer to _guarantee_ that the new params are there and supported.

So: bump the version, but don't fail if the new params are used even under 2.0. KISS.

Comment (sfoster1) [reply api/src/opentrons/protocol_api/contexts.py]: That's a good point; I personally lean towards the strict side but I wouldn't be too upset if there was just a warning.

**Pull request merged**